### PR TITLE
Issue #7588: Update doc for FinalParameters

### PR DESCRIPTION
**Pull request by ek08**
Issue #7588:Update doc for FinalParameters

### CODE REVIEW

**Comment by ek08** [comment]
#9035

**Comment by ek08** [comment]
```
$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="severity" value="error"/>
    <module name="TreeWalker">
		<module name="FinalParameters">    
     </module>
    </module>
</module>

$ cat MyClass.java
class MyClass {
  public MyClass(final int num) { // OK
  }
  public void myMethod(final String str) { // OK
  }
}

$ java -jar "-Duser.language=en -Duser.country=US" checkstyle-8.37-all.jar -c config.xml MyClass.java
Starting audit...
Audit done.

$ cat MyClass.java
class MyClass {
  public MyClass(int num) { // violation, num should be final
  }
  public void myMethod(final String str) { // OK
  }
}

$ java -jar "-Duser.language=en -Duser.country=US" checkstyle-8.37-all.jar -c config.xml MyClass.java
Starting audit...
[ERROR] /home/ehtesham/MyClass.java:2:18: Parameter num should be final. [FinalParameters]
Audit done.
Checkstyle ends with 1 errors.

$ cat config2.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="severity" value="error"/>
    <module name="TreeWalker">
		<module name="FinalParameters">
      <property name="tokens" value="CTOR_DEF"/>
     </module>
    </module>
</module>

$ cat MyClass.java
class MyClass {
  public MyClass(int num) { // violation, num should be final
  }
  public void myMethod(String str) { // OK
  }
}

$ java -jar "-Duser.language=en -Duser.country=US" checkstyle-8.37-all.jar -c config2.xml MyClass.java
Starting audit...
[ERROR] /home/ehtesham/MyClass.java:2:18: Parameter num should be final. [FinalParameters]
Audit done.
Checkstyle ends with 1 errors.

$ cat config3.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="severity" value="error"/>
    <module name="TreeWalker">
		<module name="FinalParameters">
      <property name="ignorePrimitiveTypes" value="true"/>
     </module>
    </module>
</module>

$ cat MyClass.java
class MyClass {
  public MyClass(final int num) { // OK
  }
  public void myMethod(String str) { // violation, str should be final
  }
}

$ java -jar "-Duser.language=en -Duser.country=US" checkstyle-8.37-all.jar -c config3.xml MyClass.java
Starting audit...
[ERROR] /home/ehtesham/MyClass.java:4:24: Parameter str should be final. [FinalParameters]
Audit done.
Checkstyle ends with 1 errors.

$ cat MyClass.java
class MyClass {
  public MyClass(final int num) { // OK
  }
  public void myMethod(boolean bool) { // OK, boolean is a primitive datatype
  }
}

$ java -jar "-Duser.language=en -Duser.country=US" checkstyle-8.37-all.jar -c config3.xml MyClass.java
Starting audit...
Audit done.

**Comment by romani** [comment]
Github, rebase

**Comment by ek08** [comment]
@romani please restart the semaphore CI it stopped due to some reason

**Review by pbludov — APPROVED** [review]

**Comment by romani** [comment]
Github, generate web site

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/FinalParametersCheck.java [hunk 1]
```diff
@@ -68,6 +68,24 @@
...
+ * <pre>
+ * class MyClass {
+ *   public MyClass(final int num) { // OK
+ *   }
+ *   public void myMethod(final String str) { // OK
+ *   }
+ * }
+ *
+ * class MyClass {
+ *   public MyClass(int num) { // violation, num should be final
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/FinalParametersCheck.java:82]: please move this ctor to previous class  and change type of parameter.

code in example/pre tag, should be compile-able. Two classes with same name is not ok.

in this way you will show all on single class.
And it is better to reuse the same class in all other examples to show how behavior is changing by usage of some properties.

**Review by romani — CHANGES_REQUESTED** [review]
item:

**Comment by romani** [comment]
@ek08 , ping

**Comment by ek08** [comment]
> @ek08 , ping

I'm doing it,I'm so sorry for the delay I'm having exams these days

**Comment by romani** [comment]
Github, rebase

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/FinalParametersCheck.java [hunk 2]
```diff
@@ -68,6 +68,19 @@
  * &lt;module name=&quot;FinalParameters&quot;/&gt;
  * </pre>
  * <p>
+ * Example :
+ * </p>
+ * <pre>
+ * class MyClass {
+ *   public MyClass(final int num) { // OK
+ *   }
+ *   public MyClass(long l) { // violation, l should be final
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/FinalParametersCheck.java:77]: `public MyClass(long value)`
no single char symbols.

please use this class in all examples, so only comment "//OK" and "//violation" are changing between examples

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/FinalParametersCheck.java (hunk 2 above)
Comment (ek08) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/FinalParametersCheck.java]: done

**Comment by ek08** [comment]
I'm merging the commits into one

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/FinalParametersCheck.java [hunk 3]
```diff
@@ -86,6 +110,24 @@
...
+ * <pre>
+ * class MyClass {
+ *   public MyClass(final int num) { // OK
+ *   }
+ *   public void myMethod(String str) { // violation, str should be final
+ *   }
+ * }
+ *
+ * class MyClass {
+ *   public MyClass(final int num) { // OK
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/FinalParametersCheck.java:124]: please combine this class with previous.

there should be single class in example HTML tag and the same class used for different checkstyle configs, to let user clearly see how configuration changes change behavior of the same code.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/FinalParametersCheck.java [hunk 4]
```diff
@@ -76,6 +89,17 @@
  * &lt;/module&gt;
  * </pre>
  * <p>
+ * Example :
+ * </p>
+ * <pre>
+ * class MyClass {
+ *   public MyClass(int num) { // violation, num should be final
+ *   }
+ *   public void myMethod(String str) { // OK
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/FinalParametersCheck.java:98]: one more time:
the same class (with 2 ctors and one menthod) should be used in all examples.

**Review by romani — CHANGES_REQUESTED** [review]
items:

**Comment by romani** [comment]
@ek08, please finish PR

**Comment by romani** [comment]
@ek08, ping

**Pull request closed without merging**